src/multi-column-adapter/src/main/scala/MultiColumnAdapter.scala

```diff
@@ -4,67 +4,74 @@
   */
-class MultiColumnAdapter(override val uid: String) extends Transformer with MMLParams {
+class MultiColumnAdapter(override val uid: String) extends Estimator[PipelineModel] with Wrappable with ComplexParamsWritable {
 
   def this() = this(Identifiable.randomUID("MultiColumnAdapter"))
 
-  /** Comma separated list of input column names, encoded as a string. These are the columns to be transformed.
+  /** List of input column names, encoded as a string. These are the columns for the pipeline stage.
     * @group param
     */
-  val inputCols: Param[String] =
-    StringParam(
-      this,
+  val inputCols: StringArrayParam =
+    new StringArrayParam(this,
       "inputCols",
-      "comma separated list of column names encoded as a string")
+      "list of column names encoded as a string")
 
   /** @group getParam */
-  final def getInputCols: String = $(inputCols)
+  final def getInputCols: Array[String] = $(inputCols)
 
   /** @group setParam */
-  def setInputCols(value: String): this.type = set(inputCols, value)
+  def setInputCols(value: Array[String]): this.type = set(inputCols, value)
 
-  /** Comma separated list of column names for the transformed columns, encoded as a string.
+  /** List of column names for the pipeline staged columns, encoded as a string.
     * @group param
     */
-  val outputCols: Param[String] =
-    StringParam(
-      this,
+  val outputCols: StringArrayParam =
+    new StringArrayParam(this,
       "outputCols",
-      "comma separated list of column names encoded as a string")
+      "list of column names encoded as a string")
 
   /** @group getParam */
-  final def getOutputCols: String = $(outputCols)
+  final def getOutputCols: Array[String] = $(outputCols)
 
   /** @group setParam */
-  def setOutputCols(value: String): this.type = set(outputCols, value)
+  def setOutputCols(value: Array[String]): this.type = set(outputCols, value)
 
   /** @return List of input/output column name pairs. */
-  def getInputOutputPairs: List[(String, String)] =
-    getInputCols.split(",").zip(getOutputCols.split(",")).toList
+  def getInputOutputPairs: List[(String, String)] = getInputCols.zip(getOutputCols).toList
 
-  /** Base transformer to apply to every column in the input column list.
+  def getStages: Array[PipelineStage] = getInputOutputPairs.toArray.map(getInOutPairStage)
+
+  private def getInOutPairStage(inputOutputPair: (String, String)): PipelineStage = {
+    val model = getBasePipelineStage
+    setParamInternal(setParamInternal(model,
+      "inputCol", inputOutputPair._1),
+      "outputCol", inputOutputPair._2)
+    model
+  }
+
+  /** Base pipeline stage to apply to every column in the input column list.
     * @group param
     */
-  val baseTransformer: TransformerParam =
-    new TransformerParam(this,
-                         "baseTransformer",
-                         "base transformer to apply to every column")
+  val basePipelineStage: PipelineStageParam =
```

Comment:
nit: perhaps rename to just stage, this is what the pipeline and timer APIs look like

Reply:
In the Pipeline, there is a distinction with stages. My only concern is that we use this param to initialize stages, it is not used as a stage itself. baseStage may take this into account more so than stage. What do you think?